src/App/App.js

```diff
@@ -13,7 +13,8 @@ class App extends Component {
     this.state = {
       movies: [],
       error: '',
-      selectedMovie: ''
+      selectedMovie: '',
+      location: ''
```

Comment:
This `location` holds the ID of the last movie that was searched. It is passed as a prop to the Movie component and the movie component compares it to the ID of the current movie being displayed. If the ID changes the component knows to rerender. This solves the issue of the Movie not updating despite the path in the URL changing.